### PR TITLE
Run acceptance tests on firefox

### LINKED ISSUE #1197: run UI tests in Firefox

### PR DESCRIPTION
**Pull request by paurakhsharma**
## Description
Run acceptance tests on firefox

`moveToElement()` is removed because it causes `'move target out of bounds'` on firefox.
According to documentation when running `click()` method
> `The element is scrolled into view if it is not already pointer-interactable`.

So, the tests should still function as usual.

## Related Issue
Fixes #1197

## Motivation and Context
Make the tests more robust. 

## How Has This Been Tested?
🤖

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary><p>





</p></details>

**Comment by individual-it** [comment]
FF is crashing
```
17:02:31.196 INFO [ProtocolHandshake.createSession] - Detected dialect: W3C
17:02:31.226 INFO [RemoteSession$Factory.lambda$performHandshake$0] - Started new session 7d22c1f5-6060-4604-9d48-5fec24509fd1 (org.openqa.selenium.firefox.GeckoDriverService)
JavaScript error: http://phoenix:8300/#/login, line 36: ReferenceError: wgxpath is not defined
JavaScript error: http://phoenix:8300/index.html#/oidc-callback?access_token=Y3DzV6sfH0UNb6o4cSqSJT720sYOUdOerVeR1z4c8QPxW7yxKaxaeJZxhe9I9aPx&expires_in=3600&state=adfc56985b884229a860d033c520ccfc, line 36: ReferenceError: wgxpath is not defined
JavaScript error: http://phoenix:8300/, line 36: ReferenceError: wgxpath is not defined
[Parent 252, Gecko_IOThread] WARNING: pipe error (98): Connection reset by peer: file /builds/worker/workspace/build/src/ipc/chromium/src/chrome/common/ipc_channel_posix.cc, line 357

###!!! [Parent][MessageChannel] Error: (msgtype=0x1F0088,name=PBrowser::Msg_Destroy) Channel error: cannot send/recv

A content process crashed and MOZ_CRASHREPORTER_SHUTDOWN is set, shutting down
1559581362760 Marionette INFO Stopped listening on port 34652
```

will restart the build to see how often it happens

File: tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js [hunk 1]
```diff
@@ -69,7 +69,13 @@ module.exports = {
         .waitForElementVisible('@sharingAutoCompleteDropDownElements')
         .api.elements('css selector', this.elements['sharingAutoCompleteDropDownElements'].selector, (result) => {
           result.value.forEach((value) => {
-            webElementIdList.push(value.ELEMENT)
+            // webElementId is returned as value.ELEMENT on chrome
+            // whereas it is returned as string constant "element-6066-11e4-a52e-4f735466cecf" on firefox
+            if (value.ELEMENT) {
+              webElementIdList.push(value.ELEMENT)
+            } else {
+              webElementIdList.push(value['element-6066-11e4-a52e-4f735466cecf'])
```
Comment (individual-it) [inline tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js]: is that magic number always the same?

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary><p>




</p></details>

File: tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js (hunk 1 above)
Comment (paurakhsharma) [reply tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js]: Yes, you can read more about it here: https://www.w3.org/TR/webdriver/#elements

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary><p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIRenameFiles</strong> failed. Please find the screenshots inside ...</summary><p>





</p></details>

**Comment by paurakhsharma** [comment]
There is an error in some suites with an error message 
```js
{ state: 'unknown error',
     sessionId: null,
     value: 
      { systemInformation: 'System info: host: \'606a6335e49d\', ip: \'172.17.0.10\', os.name: \'Linux\', os.arch: \'amd64\', os.version: \'4.4.0-148-generic\', java.version: \'1.8.0_212\'',
        message: 'Failed to decode response from marionette',
        error: 
         [ 'Build info: version: \'3.141.59\', revision: \'e82be7d358\', time: \'2018-11-14T08:25:53\'',
           'System info: host: \'606a6335e49d\', ip: \'172.17.0.10\', os.name: \'Linux\', os.arch: \'amd64\', os.version: \'4.4.0-148-generic\', java.version: \'1.8.0_212\'',
           'Driver info: driver.version: unknown' ] },
     status: 13 }
```

Which might be caused due to memory problem in drone,
according to this issue. https://github.com/mozilla/geckodriver/issues/1193#issuecomment-369696152

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary><p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary><p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary><p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary><p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFiles</strong> failed. Please find the screenshots inside ...</summary><p>







</p></details>

**Comment by paurakhsharma** [comment]
>  Acceptance tests **webUIFiles** failed. Please find the screenshots inside ...
> 

Another problem while running tests in firefox is that the user gets logged out when running 
the step `When the user reloads the current page of the webUI` this happens locally too.
c.c @individual-it

File: .drone.yml [hunk 2]
```diff
@@ -210,12 +212,19 @@ services:
...
+  chrome:
     image: selenium/standalone-chrome-debug:latest
     pull: true
     when:
       matrix:
-        TEST_SUITE: acceptance
+        BROWSER: chrome
+
+  firefox:
+    image: selenium/standalone-firefox-debug:3.141.59-mercury
```
Comment (individual-it) [inline .drone.yml]: why not `:latest`?

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFiles</strong> failed. Please find the screenshots inside ...</summary><p>















</p></details>

File: tests/acceptance/features/webUIFiles/createFolderEdgeCases.feature [hunk 3]
```diff
@@ -8,19 +8,19 @@ Feature: create folder
...
-    Then folder <folder_name> should be listed on the webUI
-    Examples:
-      | folder_name              |
-      | 'सिमप्ले फोल्देर $%#?&@' |
-      | '"somequotes1"'          |
-      | "'somequotes2'"          |
-      | "^#29][29@({"            |
-      | "+-{$(882)"              |
-      | "home"                   |
+#  Scenario Outline: Create a folder using special characters
```
[the remark this answers is not in the document]
Comment (paurakhsharma) [reply tests/acceptance/features/webUIFiles/createFolderEdgeCases.feature]: That's a mistake, that should have`@skipOnFIREFOX`

File: .drone.yml (hunk 2 above)
Comment (paurakhsharma) [reply .drone.yml]: Oh, I used for testing purpose. I should keep the `latest` one there.

**Comment by paurakhsharma** [comment]
> all in all this looks a bit haky to me and might make the tests more flaky in the future. Should we maybe just archive this and try again in a couple of month?

Yeah its haky, we have skipped many scenarios that are failing on firefox without knowing the actual
cause. Yes, closing this PR for now and then reopening again later when the `geckodriver` is more stable makes sense to me.

**Comment by paurakhsharma** [comment]
Closing this for now. We can retry after https://github.com/mozilla/geckodriver/issues/285 is fixed.

**Pull request closed without merging**